src/cp_gemm_interface.F

```diff
@@ -81,23 +74,16 @@ SUBROUTINE cp_gemm(transa, transb, m, n, k, alpha, matrix_a, matrix_b, beta, &
       INTEGER                                            :: handle, handle1, my_multi
       INTEGER, DIMENSION(:), POINTER                     :: a_col_loc, a_row_loc, b_col_loc, &
                                                             b_row_loc, c_col_loc, c_row_loc
-      TYPE(dbcsr_type)                                   :: a_db, b_db, c_db
 
       CALL timeset(routineN, handle)
 
       my_multi = cp_fm_get_mm_type()
 
       ! catch the special case that matrices have different blocking
-      ! SCALAPACK can deal with it but dbcsr doesn't like it
+      ! SCALAPACK/COSMA can deal with it but dbcsr doesn't like it
       CALL cp_fm_get_info(matrix_a, nrow_locals=a_row_loc, ncol_locals=a_col_loc)
       CALL cp_fm_get_info(matrix_b, nrow_locals=b_row_loc, ncol_locals=b_col_loc)
       CALL cp_fm_get_info(matrix_c, nrow_locals=c_row_loc, ncol_locals=c_col_loc)
-      IF (PRESENT(a_first_row)) my_multi = do_scalapack
```

Comment:
Don't forget to pass all those `[a,b,c]_first_[row,col]` arguments to `cosma_pdgemm()` now.

Reply:
Yes, working in progress (and good catch) ;)